### ISSUE TITLE
[Bug] Moving Searched Items to another Panel with SHIFT key moves non selected/filtered items as well to new panel

### ISSUE THREAD

**Issue by devgid** [mbnuqw/sidebery#1432]
## Steps to reproduce
- Have a setup where search bar is shown and two or more panels are set up
- Have multiple parent tabs (no nesting) open both Active and Inactive e.g. 
   - Site A Page 1, Site B Page 1, Site A Page 2, Site A Page 3, Site B Page 2
- In the Search Bar enter text Site B. You will see the two site B tabs filtered in the list
- Use Shift key to click on first and last in list to select the 2 Site B filtered items and  either a)Drag to new Panel or b) Right click - Move To Panel
- Notice that it is just not the two Site B filtered tabs that are moved to the Panel but also the unfiltered in between tabs e.g. Site A Page 2+3 is also moved to the panel despite not having been selected nor being a child tab.
(Using Shift is useful if you have multiple filtered tabs and don't want to click by one with ctrl to select multiple tabs)

## Expected behavior
Only tabs highlighted visually should be moved to the other Panel

## Actual behavior
Tabs not filtered or displayed in the filtered list are also moved to the other Panel

## Debug info

<details>
<summary><b>Addon data</b></summary>
     
```json
{
  "addonVersion": "5.0.0",
  "firefoxVersion": "122.0",
  "settings": {
    "nativeScrollbars": true,
    "nativeScrollbarsThin": true,
    "nativeScrollbarsLeft": false,
    "selWinScreenshots": false,
    "updateSidebarTitle": true,
    "markWindow": false,
    "markWindowPreface": "[Sidebery] ",
    "ctxMenuNative": false,
    "ctxMenuRenderInact": true,
    "ctxMenuRenderIcons": true,
    "ctxMenuIgnoreContainers": "",
    "navBarLayout": "horizontal",
    "navBarInline": true,
    "navBarSide": "left",
    "hideAddBtn": false,
    "hideSettingsBtn": false,
    "navBtnCount": true,
    "hideEmptyPanels": true,
    "hideDiscardedTabPanels": false,
    "navActTabsPanelLeftClickAction": "none",
    "navActBookmarksPanelLeftClickAction": "none",
    "navTabsPanelMidClickAction": "discard",
    "navBookmarksPanelMidClickAction": "none",
    "navSwitchPanelsWheel": true,
    "subPanelRecentlyClosedBar": false,
    "subPanelBookmarks": false,
    "subPanelHistory": false,
    "groupLayout": "grid",
    "containersSortByName": false,
    "skipEmptyPanels": false,
    "dndTabAct": true,
    "dndTabActDelay": 750,
    "dndTabActMod": "none",
    "dndExp": "pointer",
    "dndExpDelay": 750,
    "dndExpMod": "none",
    "dndOutside": "win",
    "dndActTabFromLink": false,
    "dndActSearchTab": false,
    "dndMoveTabs": false,
    "dndMoveBookmarks": false,
    "searchBarMode": "static",
    "searchPanelSwitch": "same_type",
    "searchBookmarksShortcut": "",
    "searchHistoryShortcut": "",
    "warnOnMultiTabClose": "collapsed",
    "activateLastTabOnPanelSwitching": true,
    "activateLastTabOnPanelSwitchingLoadedOnly": true,
    "switchPanelAfterSwitchingTab": "always",
    "tabRmBtn": "hover",
    "activateAfterClosing": "next",
    "activateAfterClosingStayInPanel": true,
    "activateAfterClosingGlobal": false,
    "activateAfterClosingNoFolded": true,
    "activateAfterClosingNoDiscarded": true,
    "askNewBookmarkPlace": true,
    "tabsRmUndoNote": true,
    "tabsUnreadMark": false,
    "tabsUpdateMark": "all",
    "tabsUpdateMarkFirst": true,
    "tabsReloadLimit": 5,
    "tabsReloadLimitNotif": true,
    "showNewTabBtns": true,
    "newTabBarPosition": "after_tabs",
    "tabsPanelSwitchActMove": false,
    "tabsPanelSwitchActMoveAuto": false,
    "tabsUrlInTooltip": "full",
    "newTabCtxReopen": false,
    "tabWarmupOnHover": true,
    "tabSwitchDelay": 0,
    "moveNewTabPin": "start",
    "moveNewTabParent": "last_child",
    "moveNewTabParentActPanel": false,
    "moveNewTab": "end",
    "moveNewTabActivePin": "start",
    "pinnedTabsPosition": "panel",
    "pinnedTabsList": false,
    "pinnedAutoGroup": false,
    "pinnedNoUnload": false,
    "tabsTree": true,
    "groupOnOpen": true,
    "tabsTreeLimit": "none",
    "autoFoldTabs": false,
    "autoFoldTabsExcept": "none",
    "autoExpandTabs": false,
    "autoExpandTabsOnNew": false,
    "rmChildTabs": "folded",
    "tabsLvlDots": true,
    "discardFolded": false,
    "discardFoldedDelay": 0,
    "discardFoldedDelayUnit": "sec",
    "tabsTreeBookmarks": true,
    "treeRmOutdent": "branch",
    "ignoreFoldedParent": false,
    "colorizeTabs": false,
    "colorizeTabsSrc": "domain",
    "colorizeTabsBranches": false,
    "colorizeTabsBranchesSrc": "url",
    "inheritCustomColor": true,
    "hideInact": false,
    "hideFoldedTabs": false,
    "hideFoldedParent": "none",
    "nativeHighlight": false,
    "warnOnMultiBookmarkDelete": "collapsed",
    "autoCloseBookmarks": false,
    "autoRemoveOther": false,
    "highlightOpenBookmarks": false,
    "activateOpenBookmarkTab": false,
    "showBookmarkLen": true,
    "bookmarksRmUndoNote": true,
    "loadBookmarksOnDemand": true,
    "pinOpenedBookmarksFolder": true,
    "oldBookmarksAfterSave": "ask",
    "loadHistoryOnDemand": true,
    "fontSize": "m",
    "animations": true,
    "animationSpeed": "norm",
    "theme": "proton",
    "density": "default",
    "colorScheme": "ff",
    "sidebarCSS": false,
    "groupCSS": false,
    "snapNotify": true,
    "snapExcludePrivate": false,
    "snapInterval": 0,
    "snapIntervalUnit": "min",
    "snapLimit": 0,
    "snapLimitUnit": "snap",
    "snapAutoExport": false,
    "snapAutoExportType": "json",
    "snapAutoExportPath": "Sidebery/snapshot-%Y.%M.%D-%h.%m.%s",
    "snapMdFullTree": false,
    "hScrollAction": "none",
    "navSwitchPanelsDelay": 128,
    "scrollThroughTabs": "none",
    "scrollThroughVisibleTabs": true,
    "scrollThroughTabsSkipDiscarded": true,
    "scrollThroughTabsExceptOverflow": true,
    "scrollThroughTabsCyclic": false,
    "scrollThroughTabsScrollArea": 0,
    "autoMenuMultiSel": true,
    "multipleMiddleClose": false,
    "longClickDelay": 500,
    "wheelThreshold": false,
    "wheelThresholdX": 10,
    "wheelThresholdY": 60,
    "tabDoubleClick": "none",
    "tabsSecondClickActPrev": true,
    "tabsSecondClickActPrevPanelOnly": false,
    "shiftSelAct": true,
    "activateOnMouseUp": false,
    "tabLongLeftClick": "none",
    "tabLongRightClick": "none",
    "tabMiddleClick": "close",
    "tabMiddleClickCtrl": "discard",
    "tabMiddleClickShift": "duplicate",
    "tabCloseMiddleClick": "close",
    "tabsPanelLeftClickAction": "none",
    "tabsPanelDoubleClickAction": "tab",
    "tabsPanelRightClickAction": "menu",
    "tabsPanelMiddleClickAction": "tab",
    "newTabMiddleClickAction": "new_child",
    "bookmarksLeftClickAction": "open_in_act",
    "bookmarksLeftClickActivate": false,
    "bookmarksLeftClickPos": "default",
    "bookmarksMidClickAction": "open_in_new",
    "bookmarksMidClickActivate": false,
    "bookmarksMidClickRemove": false,
    "bookmarksMidClickPos": "default",
    "historyLeftClickAction": "open_in_act",
    "historyLeftClickActivate": false,
    "historyLeftClickPos": "default",
    "historyMidClickAction": "open_in_new",
    "historyMidClickActivate": false,
    "historyMidClickPos": "default",
    "syncName": "",
    "syncSaveSettings": false,
    "syncSaveCtxMenu": false,
    "syncSaveStyles": false,
    "syncSaveKeybindings": false,
    "selectActiveTabFirst": true
  },
  "permissions": {
    "allUrls": true,
    "webRequest": true,
    "webRequestBlocking": true,
    "proxy": true,
    "tabHide": false,
    "clipboardWrite": false,
    "history": false,
    "bookmarks": true,
    "downloads": false
  },
  "storage": {
    "size": "1.8 mb",
    "props": {
      "containers": "1.51 kb",
      "expandedBookmarkFolders": "38 b",
      "favDomains": "68.3 kb",
      "favHashes": "16.8 kb",
      "favicons_01": "484 kb",
      "favicons_02": "484 kb",
      "favicons_03": "420 kb",
      "favicons_04": "277 kb",
      "favicons_05": "2 b",
      "lastSnapTime": "13 b",
      "profileID": "14 b",
      "settings": "4.81 kb",
      "sidebar": "1.66 kb",
      "snapshots": "2 b",
      "tabsDataCache": "86.5 kb",
      "ver": "7 b"
    }
  },
  "sidebar": {
    "nav": [
      "RhqlC5G0RE8s",
      "eU9Gf5oRDxcm",
      "oUehKQ5aOSAm",
      "ukyzA30gxOyo",
      "hdn"
    ],
    "panels": {
      "RhqlC5G0RE8s": {
        "type": 2,
        "id": "RhqlC5G0RE8s",
        "name": "len: 4",
        "color": "toolbar",
        "iconSVG": "icon_tabs",
        "iconIMGSrc": "",
        "iconIMG": "",
        "lockedPanel": true,
        "skipOnSwitching": true,
        "noEmpty": false,
        "newTabCtx": "none",
        "dropTabCtx": "none",
        "moveRules": [],
        "moveExcludedTo": -1,
        "bookmarksFolderId": -1,
        "newTabBtns": [
          ""
        ],
        "srcPanelConfig": null
      },
      "eU9Gf5oRDxcm": {
        "type": 2,
        "id": "eU9Gf5oRDxcm",
        "name": "len: 9",
        "color": "blue",
        "iconSVG": "fence",
        "iconIMGSrc": "",
        "iconIMG": "",
        "lockedPanel": true,
        "skipOnSwitching": true,
        "noEmpty": false,
        "newTabCtx": "none",
        "dropTabCtx": "none",
        "moveRules": [],
        "moveExcludedTo": -1,
        "bookmarksFolderId": -1,
        "newTabBtns": [],
        "srcPanelConfig": null
      },
      "HZVeUgjwrKim": {
        "type": 2,
        "id": "HZVeUgjwrKim",
        "name": "len: 6",
        "color": "green",
        "iconSVG": "icon_edu",
        "iconIMGSrc": "",
        "iconIMG": "",
        "lockedPanel": false,
        "skipOnSwitching": false,
        "noEmpty": false,
        "newTabCtx": "none",
        "dropTabCtx": "none",
        "moveRules": [],
        "moveExcludedTo": -1,
        "bookmarksFolderId": -1,
        "newTabBtns": [],
        "srcPanelConfig": null
      },
      "oUehKQ5aOSAm": {
        "type": 2,
        "id": "oUehKQ5aOSAm",
        "name": "len: 8",
        "color": "yellow",
        "iconSVG": "icon_books",
        "iconIMGSrc": "",
        "iconIMG": "",
        "lockedPanel": true,
        "skipOnSwitching": true,
        "noEmpty": false,
        "newTabCtx": "none",
        "dropTabCtx": "none",
        "moveRules": [],
        "moveExcludedTo": -1,
        "bookmarksFolderId": -1,
        "newTabBtns": [],
        "srcPanelConfig": null
      },
      "ukyzA30gxOyo": {
        "type": 2,
        "id": "ukyzA30gxOyo",
        "name": "len: 6",
        "color": "green",
        "iconSVG": "icon_code",
        "iconIMGSrc": "",
        "iconIMG": "",
        "lockedPanel": false,
        "skipOnSwitching": false,
        "noEmpty": false,
        "newTabCtx": "none",
        "dropTabCtx": "none",
        "moveRules": [],
        "moveExcludedTo": -1,
        "bookmarksFolderId": -1,
        "newTabBtns": [],
        "srcPanelConfig": null
      }
    }
  },
  "containers": [
    {
      "id": "firefox-container-1",
      "cookieStoreId": "firefox-container-1",
      "name": "8",
      "icon": "...",
      "color": "blue",
      "colorCode": "#37adff",
      "proxified": false,
      "proxy": null,
      "reopenRulesActive": false,
      "reopenRules": [],
      "userAgentActive": false,
      "userAgent": ""
    },
    {
      "id": "firefox-container-2",
      "cookieStoreId": "firefox-container-2",
      "name": "4",
      "icon": "...",
      "color": "orange",
      "colorCode": "#37adff",
      "proxified": false,
      "proxy": null,
      "reopenRulesActive": false,
      "reopenRules": [],
      "userAgentActive": false,
      "userAgent": ""
    },
    {
      "id": "firefox-container-3",
      "cookieStoreId": "firefox-container-3",
      "name": "7",
      "icon": "...",
      "color": "green",
      "colorCode": "#37adff",
      "proxified": false,
      "proxy": null,
      "reopenRulesActive": false,
      "reopenRules": [],
      "userAgentActive": false,
      "userAgent": ""
    },
    {
      "id": "firefox-container-4",
      "cookieStoreId": "firefox-container-4",
      "name": "8",
      "icon": "...",
      "color": "pink",
      "colorCode": "#37adff",
      "proxified": false,
      "proxy": null,
      "reopenRulesActive": false,
      "reopenRules": [],
      "userAgentActive": false,
      "userAgent": ""
    },
    {
      "id": "firefox-container-6",
      "cookieStoreId": "firefox-container-6",
      "name": "9",
      "icon": "...",
      "color": "blue",
      "colorCode": "#37adff",
      "proxified": false,
      "proxy": null,
      "reopenRulesActive": true,
      "reopenRules": [
        {
          "id": "bc2xWXJ53Icm",
          "active": true,
          "type": 1,
          "url": "/^https?://([0-9A-Za-z-]{1,63}\\.)*www\\.ixxx\\.com/",
          "name": "www.ixxx.com"
        }
      ],
      "userAgentActive": false,
      "userAgent": "",
      "includeHostsActive": false,
      "excludeHostsActive": false
    }
  ],
  "windows": [
    {
      "state": "maximized",
      "incognito": false,
      "tabsCount": 19
    },
    {
      "state": "maximized",
      "incognito": false,
      "tabsCount": 645
    },
    {
      "state": "maximized",
      "incognito": false,
      "tabsCount": 27
    }
  ],
  "bookmarks": {
    "bookmarksCount": 1793,
    "foldersCount": 68,
    "separatorsCount": 0,
    "maxDepth": 5
  }
}
```
</details>

<details>
<summary><b>Logs</b></summary>

```
[sidebar:9] Tabs.scrollToTab: Cannot find tab element [chunk-4SCIW35R.js:2:7893](moz-extension://3a517989-6b52-48aa-b612-f750c012037f/chunk-4SCIW35R.js)
```
</details>

**Comment by adhdel** [comment]
The same happens when trying to filter tabs and delete them.
1 - Search for tabs matching a keyword, e.g. a certain website. 
2 - From the filtered list, select any range of tabs with Shift, then delete
3 - Expected result: only the tabs visually selected will be deleted, only those matching the search
4 - All tabs in the selected range are deleted, not just those filtered by the search. 
For example, it seems that if I have 300 tabs, and the first filtered tab is position 100, the last filtered tab is position 200: all tabs between 100 and 200 will be deleted, not just those that matched the search. Only tabs that were before position 100 or after position 200 will remain.

**Comment by adhdel** [comment]
With 5.2.0., multi-select seems to work as intended, e.g. selecting only the filtered tabs.

**Comment by mbnuqw** [comment]
Fixed in v5.2.0
